flink-table/flink-table-planner/src/test/java/org/apache/flink/table/catalog/CatalogManagerTest.java

```diff
@@ -160,6 +167,43 @@ public void testDropTemporaryTable() throws Exception {
                 manager.qualifyIdentifier(UnresolvedIdentifier.of("test")), false);
     }
 
+    @Test
+    public void testDropTableWithWatermakForProcTime() throws Exception {
```

Comment:
A simpliest way is adding a test in `org.apache.flink.table.planner.catalog.CatalogTableITCase` where you can use SQL DDL to register tables.